### PR TITLE
Adding missing actions to the release automation

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Add missing actions to release workflow

### 🛠 Implementation details

Check the comments in the PR.

### 🧪 Testing

Do the release or do a fake one.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- ~[ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)~
- [x] PR targets the `develop` branch
- ~[ ] PR is linked to the GitHub issue it resolves~

#### Code & documentation
- ~[ ] Changelog is updated with client-facing changes~
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: .github/workflows/apk-s3-distribute.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 name: App Distribute CI
 
 on:
-  release:
-    types: [released]
+  workflow_run:
```
Comment (leandroBorgesFerreira) [inline .github/workflows/apk-s3-distribute.yml:4]: When the release is automatically created, the app is then distributed to S3

File: .github/workflows/check-docs-snippets.yml [hunk 2]
```diff
@@ -1,8 +1,10 @@
 name: Check documentation snippets
 
 on:
-  release:
-    types: [published]
+  workflow_run:
+    workflows: ["Publish API docs"]
```
Comment (leandroBorgesFerreira) [inline .github/workflows/check-docs-snippets.yml:5]: After API Docs are successfully published, this actions runs.

File: .github/workflows/publish-api-docs.yml [hunk 3]
```diff
@@ -1,8 +1,10 @@
 name: Publish API docs
 
 on:
-  release:
-    types: [released]
+  workflow_run:
+    workflows: ["Publish"]
```
Comment (leandroBorgesFerreira) [inline .github/workflows/publish-api-docs.yml:5]: After the app is published to maven central, the docs are published.

**Review by JcMinarro — APPROVED** [review]
LGTM

**Review by skydoves — APPROVED** [review]
Looks good to me!

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**